dynosaur_derive/src/lib.rs

```diff
@@ -56,45 +65,166 @@ pub fn dynosaur(
+                                        .push(parse_quote_spanned!(span=> #param_name: 'async_trait + #bounds));
+                                }
+                                GenericParam::Lifetime(param) => {
+                                    let param_name = &param.lifetime;
+                                    let span = match param.colon_token.take() {
+                                        Some(colon_token) => colon_token.span,
+                                        None => param_name.span(),
+                                    };
+                                    let bounds = mem::replace(&mut param.bounds, Punctuated::new());
+                                    where_clause_or_default(&mut sig.generics.where_clause)
+                                        .predicates
+                                        .push(parse_quote_spanned!(span=> #param: 'async_trait + #bounds));
+                                }
+                                GenericParam::Const(_) => {}
+                            }
+                        }
+
+                        for param in used_lifetimes(&item_trait.generics, &lifetimes.explicit) {
+                            let param = &param.lifetime;
+                            let span = param.span();
+                            where_clause_or_default(&mut sig.generics.where_clause)
+                                .predicates
+                                .push(parse_quote_spanned!(span=> #param: 'async_trait));
+                        }
+
+                        if sig.generics.lt_token.is_none() {
+                            sig.generics.lt_token = Some(Token![<](sig.ident.span()));
+                        }
+                        if sig.generics.gt_token.is_none() {
+                            sig.generics.gt_token = Some(Token![>](sig.paren_token.span.join()));
+                        }
+
+                        for elided in lifetimes.elided {
+                            sig.generics.params.push(parse_quote!(#elided));
+                            where_clause_or_default(&mut sig.generics.where_clause)
+                                .predicates
+                                .push(parse_quote_spanned!(elided.span()=> #elided: 'async_trait));
+                        }
+
+                        sig.generics.params.push(parse_quote!('async_trait));
+
+                        where_clause_or_default(&mut sig.generics.where_clause)
+                            .predicates
+                            .push(parse_quote! {
+                                Self: 'async_trait
+                            });
+
+                        for (i, arg) in sig.inputs.iter_mut().enumerate() {
+                            match arg {
+                                FnArg::Receiver(receiver) => {
+                                    if receiver.reference.is_none() {
+                                        receiver.mutability = None;
```

Comment:
What's going on here?

Reply:
Removed as this is async_trait's code that I'm not 100% why is there. If that is really needed we can add it later.